### PR TITLE
refactor: Stop using gArgs global in system.cpp

### PR DESCRIPTION
**Pull request by ryanofsky**
Most of the code in `util/system.cpp` that was hardcoded to use the global `ArgsManager` instance `gArgs` has been changed to stop using it (for example in https://github.com/bitcoin/bitcoin/pull/20092). But a few hardcoded references to `gArgs` remain. This commit removes the last ones so these functions aren't reading or writing global state.

Noticed these `gArgs` references while reviewing #27073

### CODE REVIEW

**Comment by TheCharlatan** [comment]
Concept ACK

**Comment by fanquake** [comment]
Concept ACK

**Comment by hebasto** [comment]
Concept ACK.

**Review by stickies-v — COMMENTED** [review]
light code review ACK 9a9d5da11

**Review by willcl-ark — APPROVED** [review]
tACK 9a9d5da11

Tidy changes to remove the last references to global  `gArgs` from util/system.cpp (besides the global declaration).

File: src/util/system.cpp [hunk 1]
```diff
@@ -1068,8 +1068,8 @@ bool ArgsManager::ReadConfigFiles(std::string& error, bool ignore_invalid_keys)
     }
 
     // If datadir is changed in .conf file:
-    gArgs.ClearPathCache();
```
Comment (stickies-v) [inline src/util/system.cpp:1071]: nit: Looks like this shouldn't have been called on `gArgs` regardless of this PR, perhaps useful to split out in separate commit? In the 3 callsites of `ArgsManager::ReadConfigFiles` I see, it is only called on `gArgs` anyway, so I don't see any actual behaviour change/potential bug because of it.

**Review by stickies-v — APPROVED** [review]
ACK 9a9d5da11

**Comment by achow101** [comment]
ACK 9a9d5da11fa6033f82dcf8e2298aee29587f5396
